### ISSUE TITLE
About test.ipynb

### ISSUE THREAD

**Issue by Hpjhpjhs** [cyclomon/3dbraingen#5]
Excuse me, it seems that the file called 'test.ipynb' cannot be open. It is appreciated if you check it. Thanks so much!

**Comment by cyclomon** [comment]
Hi,
I think there is some conflict between github and the script, so please download the file and open it with your own jupyter notebook.

Thanks!

**Comment by Hpjhpjhs** [comment]
Hi:
Thanks for your reply. I have tried to do that, but it's still not open by my own jupyter notebook. Besides, I have some questions about the training code of AlphaGAN with WGAN-GP in jupyter notebook. I think that for the training of C discriminator, there should be a new real image to input rather than the same image from another discriminator. Besides, I wrote some codes to visualize the data distribution, but it seems&nbsp;unreasonable. As a result, could you share your PCA visualization part. It is appreciated if I can receive your reply.


Best wishes
Panjian Huang




------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"cyclomon"<notifications@github.com&gt;;
发送时间:&nbsp;2020年6月23日(星期二) 中午1:46
收件人:&nbsp;"cyclomon/3dbraingen"<3dbraingen@noreply.github.com&gt;;
抄送:&nbsp;"☆随偑飘動☆"<773847441@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [cyclomon/3dbraingen] About test.ipynb (#5)





 
Hi,
 I think there is some conflict between github and the script, so please download the file and open it with your own jupyter notebook.
 
Thanks!

**Comment by cyclomon** [comment]
Hi, 
I uploaded a new Test code, so please check the code.
I tried to find my code for PCA visualization, but I lost the file when I graduated.
The code is very simple.  I just used Nilearn and followed the tutorial that Nilearn provides.

Best Regards,
Gihyun Kwon

**Comment by Hpjhpjhs** [comment]
Hi:
Thanks so much for your reply. The test.ipynb can be opened now. So you mean the PCA visualization code can be found from the Nilearn package? It's appreciated if I can receive your reply.


Best wishes
Panjian Huang




------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"cyclomon"<notifications@github.com&gt;;
发送时间:&nbsp;2020年6月24日(星期三) 中午11:20
收件人:&nbsp;"cyclomon/3dbraingen"<3dbraingen@noreply.github.com&gt;;
抄送:&nbsp;"☆随偑飘動☆"<773847441@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [cyclomon/3dbraingen] About test.ipynb (#5)





 
Hi,
 I uploaded a new Test code, so please check the code.
 I tried to find my code for PCA visualization, but I lost the file when I graduated.
 The code is very simple.  I just used Nilearn and followed the tutorial that Nilearn provides.
 
Best Regards,
 Gihyun Kwon

**Comment by cyclomon** [comment]
Hi,

I'm glad to hear that the test code works now.
For PCA visualization, code is not from Nilearn, but Sklearn(Scikit-learn).
You will easily find easy sample codes for PCA.
I'm sorry that I wrongly replied.

Best regards,
Gihyun Kwon

**Comment by Hpjhpjhs** [comment]
Hi:
I have tried to apply Sklearn for visualization, but the result seems unreasonable. Whether you do height x width x depth as the features, so the input of PCA is a 2D matrix: 210 x (height x width x depth). I will still try it again. It's appreciated If you can help me when I do that in some problems. Thanks so much.


Best wishes
Panjian Huang&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"cyclomon"<notifications@github.com&gt;;
发送时间:&nbsp;2020年6月29日(星期一) 下午4:24
收件人:&nbsp;"cyclomon/3dbraingen"<3dbraingen@noreply.github.com&gt;;
抄送:&nbsp;"☆随偑飘動☆"<773847441@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [cyclomon/3dbraingen] About test.ipynb (#5)





 
Hi,
 
I'm glad to hear that the test code works now.
 For PCA visualization, code is not from Nilearn, but Sklearn(Scikit-learn).
 You will easily find easy sample codes for PCA.
 I'm sorry that I wrongly replied.
 
Best regards,
 Gihyun Kwon

**Comment by Hpjhpjhs** [comment]
Hi:
Sorry for interrupting again. I have mainly some questions about the AlphaGAN with WGAN-GP. These questions all focus on this model training.
1. You use the mean as the loss of Encoder and Generator. But when the model was training, I found that at the beginning, their loss is positive, after that, their loss will be changed to negative, will this affect the training? Furthermore, the loss of discriminator will be negative all the time. about this?


2. I studied the original paper about AlphaGAN, but comparing with your codes, I don't know whether there need to input a new real image in each neural network (Encoder, Generator, Discriminator, Code Discriminator), or just input a same real image to the four neural network.


3. I found out some other's codes about AlphaGAN, but there are different orders of training. For example, in your code, the encoder and generator are trained firstly, and then the discriminator, final training is the code discriminator, but some others are code discriminator is trained firstly. So will the order of training affect the model performance?


It's very appreciated if I can receive your reply. I know here may be a lot of problems, but they are very important to me. So if there is anything unclear here, please let me know. Looking forward to receiving your reply. Thanks so much.


Best wishes
Panjian Huang




------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"cyclomon"<notifications@github.com&gt;;
发送时间:&nbsp;2020年6月29日(星期一) 下午4:24
收件人:&nbsp;"cyclomon/3dbraingen"<3dbraingen@noreply.github.com&gt;;
抄送:&nbsp;"☆随偑飘動☆"<773847441@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [cyclomon/3dbraingen] About test.ipynb (#5)





 
Hi,
 
I'm glad to hear that the test code works now.
 For PCA visualization, code is not from Nilearn, but Sklearn(Scikit-learn).
 You will easily find easy sample codes for PCA.
 I'm sorry that I wrongly replied.
 
Best regards,
 Gihyun Kwon

**Comment by cyclomon** [comment]
1. WGAN loss value can be negative.
2. As I know of, there might be slight change in putting new inputs, but not major changes. Since I didn't tried many settings, it is not sure.
3. Training order does not affect the results.

GAN is quite sensitive to minor settings, so there will be changes in performance with changing those settings. 

Best Regards,
Gihyun Kwon

**Comment by Hpjhpjhs** [comment]
Hi:
Thanks for your reply. The first question I want to focus on AlphaWGAN. So you mean that&nbsp;
1.&nbsp;[10/200000] D: 4.58     En_Ge: -29.6    Code: 7.87  
2. [20/200000] D: -1.33    En_Ge: -52.2    Code: 8.39 
3.&nbsp;[30/200000] D: -2.71    En_Ge: -60.5    Code: 8.92 
are reasonable?


2.Since I tried to run your code about AlphaWGAN, and there would be the issues about inplace oeration. So I have to input a new image in each neural network.


3. The new question is that whether there need to add output = h5.view(h5.size()[0], -1) rather than output = h5 in discriminator network on AlphaWGAN.
4. Whether the generator needs to iterate many times than other models on AlphaWGAN. For example, set g_iter = 5, d_iter = 1, cd_iter = 1.


Looking forward to receiving your reply.


Best wishes
Panjian Huang




------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"cyclomon"<notifications@github.com&gt;;
发送时间:&nbsp;2020年7月9日(星期四) 晚上8:12
收件人:&nbsp;"cyclomon/3dbraingen"<3dbraingen@noreply.github.com&gt;;
抄送:&nbsp;"☆随偑飘動☆"<773847441@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [cyclomon/3dbraingen] About test.ipynb (#5)





  
WGAN loss value can be negative.
 
As I know of, there might be slight change in putting new inputs, but not major changes. Since I didn't tried many settings, it is not sure.
 
Training order does not affect the results.
  
GAN is quite sensitive to minor settings, so there will be changes in performance with changing those settings.
 
Best Regards,
 Gihyun Kwon
